### PR TITLE
added GENERIC_LOGIN_REQUIRED setting for a full authenticated graphite site

### PR DESCRIPTION
**Pull request by toni-moreno**
I've added a new setting to enable a full authenticated site, related to issue  https://github.com/graphite-project/graphite-web/issues/681

Is only tested with internal django user handler, not tested yet with remote or ldap authentication.

Also I've added a new look&feel for the login form: 

The old login form was:

image

And the new one:

image

### CODE REVIEW

**Comment by changbl** [comment]
No one bother to review this commit and merge it?

**Comment by toni-moreno** [comment]
Yes !!! It would be great !!!!

**Comment by tony-dillon** [comment]
Hey Toni-Moreno,

Do you think you could fix the merge conflicts?  I would love to get this merged in!

Thank you for your efforts.

File: webapp/graphite/settings.py [hunk 1]
```diff
@@ -200,6 +204,14 @@
 if USE_LDAP_AUTH and LDAP_URI is None:
   LDAP_URI = "ldap://%s:%d/" % (LDAP_SERVER, LDAP_PORT)
 
+LOGIN_EXEMPT_URLS = (
+ r'^render',
+ r'^metrics/'
```
Comment (micolous) [inline webapp/graphite/settings.py]: I think this means that data can still be accessed, just the standard web interface is unavailable.  Is there a way to protect these endpoints too?

**Pull request closed without merging**